test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -329,6 +330,187 @@ def test_shard_parameter_errors(self):
+            placements=ranks,
+        )
+        ranks[0], ranks[1], ranks[2], ranks[3] = ranks[3], ranks[0], ranks[1], ranks[2]
+        reshard_spec = ChunkShardingSpec(
+            dim=1,
+            placements=ranks,
+        )
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [13, 21])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [14, 23])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [15, 26])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [12, 24])
+        spec.dim = 1
+        reshard_spec.dim = 0
+        torch.manual_seed(0)
+        ranks[0], ranks[1] = ranks[1], ranks[0]
+        spec.placements = ranks
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [14, 23])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [15, 26])
+        spec.dim = 0
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [13, 21])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [12, 24])
+        spec.dim = 1
+        reshard_spec.dim = 1
+        ranks[2], ranks[3] = ranks[3], ranks[2]
+        spec.placements = ranks
+        ranks[1], ranks[2] = ranks[2], ranks[1]
+        reshard_spec.placements = ranks
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [13, 21])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [14, 23])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [15, 26])
+        self._run_sharded_tensor_reshard(spec, reshard_spec, [12, 24])
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_reshard_errors(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+        enumerable_sharding_spec = EnumerableShardingSpec(
+            [
+                ShardMetadata(
+                    shard_offsets=[0, 0],
+                    shard_sizes=[5, 5],
+                    placement="rank:0/cuda:0",
+                ),
+                ShardMetadata(
+                    shard_offsets=[5, 0],
+                    shard_sizes=[5, 5],
+                    placement="rank:1/cuda:1",
+                ),
+            ]
+        )
+        st = _sharded_tensor.rand(spec, 24, 12)
+        with self.assertRaisesRegex(
+            NotImplementedError, "Only ChunkShardingSpec supported for reshard."
+        ):
+            st.reshard(enumerable_sharding_spec)
+
+    def _run_partial_tensor_n_reshard(
+        self, reshard_spec, input_size, world_size, ranks, aggr_op
+    ):
+        results = []
+        results_compare = []
+        for _ in range(0, world_size):
+            tensor = torch.rand(*input_size).cuda(self.rank)
+            results.append(tensor)
+            results_compare.append(tensor.clone().detach())
+        pg = dist.new_group(ranks=ranks)
```

Comment:
Why do we need a new_group? We can just use the default group since we are anyways using all ranks?

Reply:
I see. Sure.